Minor tweaks to cosmosis interface

joezuntz opened the pull request:
- saves the data vector and theory prediction as well as just the likelihood (handy for plotting and sanity checks)
- tweak error messages on import failure or missing data set

This is the version I am currently testing in this PR:
https://github.com/joezuntz/cosmosis-standard-library/pull/173

Lbalkenhol commented:
Yes that sounds perfect!

The cost of recalculating model spectra depends on the likelihood. For primary CMB it's pretty cheap so no issue there, whereas for some lensing likelihoods there can be some bigger matrix multiplications. That being said it will still be sub-dominant to CAMB/CLASS.

joezuntz commented:
Okay, great - I've updated this to use the `likelihood_only` parameter.  I also realised what I had before was broken for non-lensing likelihoods, so I've updated that.

Finally, I realised I'd like to propose changing "nuisance_parameters" to "candl_nuisance_parameters" in the section name, because every different kind of data set has its own nuisance parameters, so that's a bit too vague. Is that okay?
